internal/provider/resource_integration.go

```diff
@@ -0,0 +1,498 @@
+		} else {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to delete integration with id %s: %s", data.ID, err.Error()))
+		}
+
+		return
+	}
+
+	tflog.Debug(ctx, "deleted integration", map[string]interface{}{
+		"id":       data.ID.ValueString(),
+		"response": messageResponse.GetMessage(),
+	})
+}
+
+func (r *integrationResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	integrationId := req.ID
+	tflog.Debug(ctx, "importing integration", map[string]interface{}{
+		"id": integrationId,
+	})
+
+	integration, _, err := r.provider.client.IntegrationsApi.GetIntegrationV2(ctx, integrationId).
+		Execute()
+	if err != nil {
+		if apiError, ok := err.(*apono.GenericOpenAPIError); ok {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to get integration with id %s, error: %s, body: %s", integrationId, apiError.Error(), string(apiError.Body())))
+		} else {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to get integration with id %s: %s", integrationId, err.Error()))
+		}
+
+		return
+	}
+
+	model, diagnostics := r.convertToModel(ctx, integration)
+	if len(diagnostics) > 0 {
+		resp.Diagnostics.Append(diagnostics...)
+		return
+	}
+
+	// Save imported data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+
+	tflog.Debug(ctx, "imported integration", map[string]interface{}{
+		"id": integrationId,
+	})
+}
+
+func (r *integrationResource) ValidateConfig(ctx context.Context, req resource.ValidateConfigRequest, resp *resource.ValidateConfigResponse) {
+	if r.provider == nil {
+		return
+	}
+
+	var model integrationResourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	config, _, err := r.provider.client.IntegrationsApi.GetIntegrationConfig(ctx, model.Type.ValueString()).
+		Execute()
+	if err != nil {
+		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Failed to get config: %s", err.Error()))
+		return
+	}
+
+	if config.GetRequiresSecret() {
+		var supportedSecretExpressions []path.Expression
+		for _, secretType := range config.SupportedSecretTypes {
+			if attributeName, ok := secretTypeAttributeNames[secretType]; ok {
+				supportedSecretExpressions = append(supportedSecretExpressions, path.MatchRoot(attributeName))
+			}
+		}
+
+		resourcevalidator.ExactlyOneOf(supportedSecretExpressions...).ValidateResource(ctx, req, resp)
+	}
+
+	metadataElements := model.Metadata.Elements()
+	for _, param := range config.GetParams() {
+		paramName := param.GetId()
+		paramPossibleValues := param.GetValues()
+		paramDefaultValue := param.GetDefault()
+
+		attributePath := path.Root(fmt.Sprintf(`metadata["%s"]`, paramName))
+
+		metadataValue, hasValue := metadataElements[paramName]
+		if !hasValue {
+			if paramDefaultValue != "" {
+				resp.Diagnostics.AddAttributeError(
```

Comment:
Why do we send an error if we have a default value?

Reply:
can't we use the default value?

Reply:
No, terraform will complain about diff between the user params and the actual params